pkg/operator/operator.go

```diff
@@ -194,12 +103,10 @@ type Reporting struct {
 	importers   map[string]*prestostore.PrometheusImporter
 }
 
-func New(logger log.FieldLogger, cfg Config) (*Reporting, error) {
-	if err := cfg.APITLSConfig.Valid(); err != nil {
-		return nil, err
-	}
-	if err := cfg.MetricsTLSConfig.Valid(); err != nil {
-		return nil, err
+func New(logger log.FieldLogger, cfg Config) (ReportingOperator, error) {
+	errs := IsValidConfig(&cfg)
+	if errs != nil {
+		return nil, errs
 	}
```

Comment:
Building off my comment from above, I wonder if we should be doing all of the reporting configuration validation in the cmd/reporting-operator driver function that provides the entry point for this operator, rather than in the constructor.

Reply:
it can be called from the `cmd` package but it should not be done there.  Commands in Kube/OpenShift follow a pattern of `Complete`, `Validate`, `Run`.  We should refactor ours to match and call the `IsValidConfig` method in the command's `Validate` method.  However we should keep it in new as well to protect against programmatic access to the operator no matter how it's started in the future.

Reply:
Example of the recommended pattern: https://github.com/openshift/oc/blob/a695d74ef1aee9a3f605d38dd8b6fae2062b63fc/pkg/cli/tag/tag.go#L93-L116